api/controllers/v1/cave/update.js

```diff
@@ -13,37 +12,50 @@ const { toCave } = require('../../../services/mapping/converters');
 const { checkRight } = sails.helpers;
 
 module.exports = async (req, res) => {
-  // Check right
-  const hasRight = await checkRight
-    .with({
-      groups: req.token.groups,
-      rightEntity: RightService.RightEntities.CAVE,
-      rightAction: RightService.RightActions.EDIT_ANY,
-    })
-    .intercept('rightNotFound', () =>
-      res.serverError(
-        'A server error occured when checking your right to update a cave.'
-      )
-    );
-  if (!hasRight) {
-    return res.forbidden('You are not authorized to update a cave.');
-  }
+  try {
+    const hasRight = await checkRight
+      .with({
+        groups: req.token.groups,
+        rightEntity: RightService.RightEntities.CAVE,
+        rightAction: RightService.RightActions.EDIT_ANY,
+      })
+      .intercept('rightNotFound', () =>
+        res.serverError(
+          'A server error occured when checking your right to update a cave.'
+        )
+      );
+    if (!hasRight) {
+      return res.forbidden('You are not authorized to update a cave.');
+    }
 
-  // Check if cave exists
-  const caveId = req.param('id');
-  const currentCave = await TCave.findOne(caveId);
-  if (!currentCave) {
-    return res.notFound({
-      message: `Cave with id ${caveId} not found.`,
-    });
-  }
+    const caveId = req.param('id');
+    const rawCave = await TCave.findOne(caveId);
+    if (!rawCave || rawCave.isDeleted) {
+      return res.notFound({
+        message: `Cave with id ${caveId} not found.`,
+      });
+    }
 
-  const cleanedData = CaveService.getConvertedDataFromClient(req);
+    const newLatitude = req.param('latitude');
+    const newLongitude = req.param('longitude');
+    const newDepth = req.param('depth');
+    const newLength = req.param('length');
+    const newTemperature = req.param('temperature');
+    const newIsDiving = req.param('isDiving');
 
-  try {
-    const updatedCave = await TCave.updateOne({
-      id: caveId,
-    }).set(cleanedData);
+    const updatedFields = {
+      reviewer: req.token.id,
+      // dateReviewed will be updated automaticly by the SQL historisation trigger
+    };
+
+    if (newLatitude) updatedFields.latitude = newLatitude;
+    if (newLongitude) updatedFields.longitude = newLongitude;
+    if (newDepth) updatedFields.depth = newDepth;
+    if (newLength) updatedFields.caveLength = newLength;
+    if (newTemperature) updatedFields.temperature = newTemperature;
+    if (newIsDiving) updatedFields.isDiving = newIsDiving;
```

Comment:
You can use the following pattern here : https://stackoverflow.com/a/40560953/16939610

```js
updatedFields = {
   ...(newLatitude && {latitude: newLatitude}),
   ...(newLongitude&& {longitude: newLongitude}
   // etc.
}
```

It's pretty much the same. I just don't like very much using if blocks (thinking in a "functional programming" way). As But as you want, I don't really mind it.